internal/diag/diag.go

```diff
@@ -3,28 +3,91 @@ package diag
 import (
 	"fmt"
 	"go/ast"
+	"go/token"
+	"slices"
 
 	"golang.org/x/tools/go/analysis"
+
+	"github.com/manuelarte/funcorder/internal/astutils"
 )
 
-func NewConstructorNotAfterStructType(structSpec *ast.TypeSpec, constructor *ast.FuncDecl) analysis.Diagnostic {
+func NewConstructorNotAfterStructType(fset *token.FileSet,
+	structSpec *ast.TypeSpec, constructor *ast.FuncDecl,
+) (analysis.Diagnostic, error) {
+	suggestedFixConstructorByte, err := astutils.NodeToByteArray(fset, constructor)
+	if err != nil {
+		return analysis.Diagnostic{}, err
+	}
+	removingPos := constructor.Pos()
+	if constructor.Doc != nil {
+		removingPos = constructor.Doc.Pos()
+	}
 	return analysis.Diagnostic{
 		Pos: constructor.Pos(),
-		Message: fmt.Sprintf("function %q for struct %q should be placed after the struct declaration",
+		Message: fmt.Sprintf("constructor %q for struct %q should be placed after the struct declaration",
 			constructor.Name, structSpec.Name),
-	}
+		URL: "https://github.com/manuelarte/funcorder?tab=readme-ov-file#check-constructors-functions-are-placed-after-struct-declaration", //nolint: lll // url
```

Comment:
https://golangci-lint.run/usage/false-positives/#nolint-directive

```suggestion
		URL: "https://github.com/manuelarte/funcorder?tab=readme-ov-file#check-constructors-functions-are-placed-after-struct-declaration", //nolint:lll // url
```